### PR TITLE
work in progress

### PR DESCRIPTION
**Pull request by drupol**
This PR will be merged in https://github.com/NixOS/nixpkgs/pull/225401 when it's done.

The idea is to create `setup hooks`.

Current status: "I don't know what I'm doing"

image

### CODE REVIEW

File: pkgs/build-support/php/build-php.nix [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+
+        runHook postInstall
+      '';
+
+      strictDeps = true;
+
+      outputHashMode = "recursive";
+      outputHashAlgo = if (finalAttrs ? vendorHash && finalAttrs.vendorHash != "") then null else "sha256";
+      outputHash = finalAttrs.vendorHash or "";
+    });
```
Comment (jtojnar) [inline pkgs/build-support/php/build-php.nix:60]: Can you use `buildComposerHooks { inherit (finalAttrs) vendorHash php composer composerLock; }` here?

Comment (drupol) [reply pkgs/build-support/php/build-php.nix:60]: yes. I merged this in the original PR, let's continue there.

Comment (drupol) [reply pkgs/build-support/php/build-php.nix:60]: I pushed the changes, but I couldn't implement it exactly as you wanted. Let me know if it's ok.  Relevant commit: https://github.com/NixOS/nixpkgs/pull/225401/commits/4a484ee2d49aa7db04f37c2eb24474834bcc0a72

**Comment by drupol** [comment]
Merged in https://github.com/NixOS/nixpkgs/pull/225401

**Pull request closed without merging**